packages/calcite-components/src/controllers/useSetFocus.ts

```diff
@@ -0,0 +1,85 @@
+import { makeGenericController } from "@arcgis/lumina/controllers";
+import { LitElement } from "@arcgis/lumina";
+import { componentFocusable } from "../utils/component";
+import { FocusableElement, focusElement, getRootNode } from "../utils/dom";
+import { type InteractiveComponent } from "../utils/interactive";
+
+type FocusStrategy = "focusable" | "tabbable";
+type FocusConfig = { target: FocusableElement; includeContainer?: boolean; strategy?: FocusStrategy };
+
+export interface UseSetFocus {
+  (getFocusTarget: () => FocusableElement | FocusConfig | undefined): Promise<void>;
+}
+
+interface SetFocusComponent extends LitElement, Partial<Pick<InteractiveComponent, "disabled">> {
+  /** Sets focus on the fist focusable `calcite-radio-button` element in the component. */
+  setFocus: () => Promise<void>;
+}
+
+/**
+ * A controller for centralized setFocus behavior.
+ *
+ * @param options
+ */
+export const useSetFocus = <T extends SetFocusComponent>(): ReturnType<
+  typeof makeGenericController<UseSetFocus, T>
+> => {
+  return makeGenericController<UseSetFocus, T>((component, controller) => {
+    let abortController: AbortController;
+
+    function handleFocusOut(): void {
+      abortController?.abort();
+    }
+
+    controller.onLoad(() => {
+      component.listen("focus", () => {
+        abortController = new AbortController();
+        component.el.addEventListener("focusout", handleFocusOut, { signal: abortController.signal });
+      });
+    });
+
+    controller.onDisconnected(() => {
+      component.el.removeEventListener("focusout", handleFocusOut);
+    });
+
+    return async (getFocusTarget): Promise<void> => {
+      if (component.disabled) {
```

Comment:
We should probably update the controller's types to expect `disabled` to be defined on the component? ☝️

Reply:
Looks like it is handled on line 14 above, is that right?

Reply:
Correct. It's leveraging the interface from the interactivity-handling module.